src/binary/ibc-relayer/commands/start.ts

```diff
@@ -14,6 +17,55 @@ import { resolveKeyFileOption } from '../../utils/options/shared/resolve-key-fil
 import { resolveMnemonicOption } from '../../utils/options/shared/resolve-mnemonic-option';
 import { signingClient } from '../../utils/signing-client';
 
+function resolveHeights(
+  {
+    scanFromSrc,
+    scanFromDest,
+    home,
+  }: {
+    scanFromSrc: number | null;
+    scanFromDest: number | null;
+    home: string;
+  },
+  logger: Logger
+): RelayedHeights | null {
+  if (!scanFromSrc && scanFromDest) {
+    throw new InvalidOptionError(
+      `You have defined "scanFromDest" but no "scanFromSrc". Both or none "scanFromSrc" and "scanFromDest" must be present.`
+    );
+  }
+
+  if (scanFromSrc && !scanFromDest) {
+    throw new InvalidOptionError(
+      `You have defined "scanFromSrc" but no "scanFromDest". Both or none "scanFromSrc" and "scanFromDest" must be present.`
+    );
+  }
+
+  if (scanFromSrc && scanFromDest) {
+    logger.info('Use heights from the command line arguments.');
+    return {
+      packetHeightA: scanFromSrc,
+      ackHeightA: scanFromSrc,
+      packetHeightB: scanFromDest,
+      ackHeightB: scanFromDest,
+    };
+  }
+
+  const lastQueriedHeightsFilePath = path.join(home, lastQueriedHeightsFile);
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const heights = require(lastQueriedHeightsFilePath);
+    logger.info(
+      `Use last queried heights from ${lastQueriedHeightsFilePath} file.`
+    );
+    return heights;
+  } catch {
+    // log?
```

Comment:
Should we log at this point? What message would be appropriate? (no heights are specified)

Reply:
Log something like `scanning entire history for packets... this may take some time`